src/shared/charts/tradeChartData.ts

```diff
@@ -26,43 +31,65 @@ export function getTradeEntries(dataset: PairHistory, filteredTrades: Trade[]) {
   // 4: color
   // 5: label
   // 6: tooltip
-
   for (let i = 0, len = filteredTrades.length; i < len; i += 1) {
     const trade: Trade = filteredTrades[i];
     if (
       trade.open_timestamp >= dataset.data_start_ts &&
       trade.open_timestamp <= dataset.data_stop_ts
     ) {
-      // Trade entry
-      tradeData.push([
-        roundTimeframe(dataset.timeframe_ms ?? 0, trade.open_timestamp),
-        trade.open_rate,
-        ENTRY_SYMB,
-        trade.is_short ? 180 : 0,
-        // (trade.profit_abs ?? 0) > 0 ? '#31e04b' : '#fc0505',
-        trade.is_short ? '#AD00FF' : '#0066FF',
-        '',
-        // trade.profit_abs,
-        buildToolTip(trade, 'entry'),
-      ]);
-    }
-    if (
-      trade.close_timestamp !== undefined &&
-      trade.close_timestamp <= dataset.data_stop_ts &&
-      trade.close_timestamp > dataset.data_start_ts
-    ) {
-      if (trade.close_date !== undefined && trade.close_rate !== undefined) {
-        // Trade exit
-        tradeData.push([
-          roundTimeframe(dataset.timeframe_ms ?? 0, trade.close_timestamp),
-          trade.close_rate,
-          EXIT_SYMB,
-          trade.is_short ? 180 : 0,
-          trade.is_short ? '#AD00FF' : '#0066FF',
-          // (trade.profit_abs ?? 0) > 0 ? '#31e04b' : '#fc0505',
-          formatPercent(trade.profit_ratio, 2),
-          buildToolTip(trade, 'exit'),
-        ]);
+      if (trade.orders) {
+        for (let i = 0; i < trade.orders.length; i++) {
+          const order: Order = trade.orders[i];
+          if (
+            order.order_filled_timestamp &&
+            order.order_filled_timestamp <= dataset.data_stop_ts &&
+            order.order_filled_timestamp > dataset.data_start_ts
+          ) {
+            // Trade entry
+            if (i === 0) {
+              tradeData.push([
+                roundTimeframe(dataset.timeframe_ms ?? 0, trade.open_timestamp),
+                order.safe_price,
+                OPEN_CLOSE_SYMBOL,
+                order.ft_order_side == 'sell' ? 180 : 0,
+                trade.is_short ? SHORT_COLOR : LONG_COLOR,
+                trade.is_short ? 'Short' : 'Long',
+                buildToolTip(trade, 'entry'),
+              ]);
+              // Trade exit
+            } else if (i === trade.orders.length - 1 && trade.close_timestamp) {
+              if (
+                trade.close_timestamp <= dataset.data_stop_ts &&
+                trade.close_timestamp > dataset.data_start_ts &&
+                trade.close_date !== undefined &&
+                trade.close_rate !== undefined
```

Comment:
I think a better check for "exit" is if it's the last order (above) - and `is_open === false`